app/components/polls/polls-edit/polls-edit-directive.js

```diff
@@ -16,60 +16,113 @@ angular.module('polls.edit.directive', [
     value: 'Image'
   }
 ])
-  .directive('pollsEdit', function (routes) {
-    return {
-      templateUrl: routes.COMPONENTS_URL + 'polls/polls-edit/polls-edit.html',
-      controller: function (_, $location, $q, $routeParams, $scope, $window, Poll) {
-        // populate model for use
-        if ($routeParams.id === 'new') {
-          $scope.model = Poll.$build();
-          $scope.isNew = true;
-        } else {
-          $scope.model = Poll.$find($routeParams.id);
+.directive('pollsEdit', function (routes) {
+  return {
+    templateUrl: routes.COMPONENTS_URL + 'polls/polls-edit/polls-edit.html',
+    controller: function (_, $http, $location, $q, $routeParams, $scope, Poll) {
+      // populate model for use
+      if ($routeParams.id === 'new') {
+        $scope.model = Poll.$build();
+        $scope.isNew = true;
+      } else {
+        $scope.model = Poll.$find($routeParams.id);
+        $scope.poll = Poll.$find($routeParams.id);
+      }
+
+      window.onbeforeunload = function (e) {
+        if(!_.isEmpty($scope.model.$dirty()) || $scope.isNew || $scope.needsSave) {
+          // show confirmation alert
+          return 'You have unsaved changes.';
         }
+      };
 
-        window.onbeforeunload = function (e) {
-          if(!_.isEmpty($scope.model.$dirty()) || $scope.isNew || $scope.needsSave) {
-            // show confirmation alert
-            return 'You have unsaved changes.';
-          }
-        };
+      $scope.$on('$destroy', function () {
+        // remove alert when we go
+        delete window.onbeforeunload;
+      });
 
-        $scope.$on('$destroy', function () {
-          // remove alert when we go
-          delete window.onbeforeunload;
-        });
+      var answerUrl = '/cms/api/v1/answer/';
 
       $scope.saveModel = function () {
         if ($scope.model) {
+          $scope.deleteSodaheadAnswers();
```

Comment:
this method will only delete answers if they have been added to the deletedAnswers array, otherwise nothing happens